src/core/config/local_config.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the ""License"");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an ""AS IS"" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { IsEmail, IsNotEmpty, IsObject, IsString, validateSync } from 'class-validator'
+import { type ListrTask } from 'listr2'
+import fs from 'fs'
+import * as yaml from 'yaml'
+import { type ClusterMapping, type Deployment, type Deployments, type LocalConfigData } from './local_config_data.js'
+import { MissingArgumentError, SoloError } from '../errors.js'
+import { promptDeploymentClusters, promptDeploymentName, promptUserEmailAddress } from '../../commands/prompts.js'
+import { flags } from '../../commands/index.js'
+import { type SoloLogger } from '../logging.js'
+import { Task } from '../task.js'
+
+export class LocalConfig implements LocalConfigData {
+    @IsNotEmpty()
+    @IsEmail()
+    userEmailAddress: string
+
+    // The string is the name of the deployment, will be used as the namespace,
+    // so it needs to be available in all targeted clusters
+    @IsNotEmpty()
+    @IsObject()
+    deployments: Deployments
+
+    @IsNotEmpty()
+    @IsString()
+    currentDeploymentName : string
+
+    // contextName refers to the "CURRENT NAME", and clusterName refers to the CLUSTER leveraged in kubeConfig.currentContext
+    // { clusterName : string, contextName : string }
+    @IsNotEmpty()
+    @IsObject()
+    clusterMappings: ClusterMapping
+
+    private readonly skipPromptTask: boolean = false
+    private readonly filePath: string
+    private readonly logger: SoloLogger
```

Comment:
move filePath and logger to constructor, see other examples in codebase

Reply:
fixed